### PR TITLE
Recipe for network testing

### PR DESCRIPTION
**Pull request by egieseke**
## Summary

Added a new recipe for performing network disruption tests.

## Test Plan

Testing using Algonet.

### CODE REVIEW

File: test/testdata/deployednettemplates/recipes/network-partition/Makefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+PARAMS=-w 100 -R 8 -N 20 -n 100 -H 10 --node-template node.json --relay-template relay.json --non-participating-node-template nonPartNode.json
```
Comment (tsachiherman) [inline test/testdata/deployednettemplates/recipes/network-partition/Makefile:1]: Do we really need such a big network to conduct a partitioning test ?
I don't really see how the large network would help us getting the answer "does the network recover after partitioning? and after how long ?"

Comment (egieseke) [reply test/testdata/deployednettemplates/recipes/network-partition/Makefile:1]: We can adjust the network to be smaller.  The github tickets ask for all 3 scenarios.

Comment (tsachiherman) [reply test/testdata/deployednettemplates/recipes/network-partition/Makefile:1]: That's true. It's that I just don't want you to get delayed by "deployment issues" that aren't related to the actual test you're trying to conduct.. Having the feedback that it's broken on small network is more valuable than the fact the it's broken on a large network ( and would probably be easier to evaluate )

Comment (egieseke) [reply test/testdata/deployednettemplates/recipes/network-partition/Makefile:1]: OK, we will start testing tomorrow.

File: netdeploy/remote/deployedNetwork.go [hunk 2]
```diff
@@ -404,6 +405,14 @@ func (cfg DeployedNetwork) GenerateCloudTemplate(templates HostTemplates, target
 		if err != nil {
 			return
 		}
+
+		group := strings.TrimSpace(cloudHost.Group)
+		if group == "" {
+			hostSpec.Group = "default"
```
Comment (tsachiherman) [inline netdeploy/remote/deployedNetwork.go:411]: For backward compatibility, you might want to leave the default as an empty string.
This would force the cloudspec file consumer to populate it with "default" instead of expecting a non-empty string.
( i.e. currently, all cloud-spec files doesn't have this field. )

If you're going to break backward compatibility, please add versioning to the cloud spec file so we can detect that forward.

File: test/testdata/deployednettemplates/recipes/network-partition/gen_topology.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        ranges[name] = {"start": start_pos, "end": end_pos}
+        start_pos = end_pos + 1
+    print(ranges)
+    return ranges
+
+ranges = get_ranges()
+
+# for i in range(10):
+#     partition_name = get_partition(ranges)
+#     print(i, partition_name)
```
Comment (tsachiherman) [inline test/testdata/deployednettemplates/recipes/network-partition/gen_topology.py:70]: nit: delete deadcode

File: docker/build/Dockerfile-deploy [hunk 4]
```diff
@@ -6,7 +6,7 @@ WORKDIR /root
 RUN wget --quiet https://dl.google.com/go/go${GOLANG_VERSION}.linux-amd64.tar.gz && tar -xvf go${GOLANG_VERSION}.linux-amd64.tar.gz && mv go /usr/local
 ENV GOROOT=/usr/local/go \
     GOPATH=$HOME/go \
-    GOPROXY=https://gocenter.io,https://goproxy.io,direct
+    GOPROXY=https://gocenter.io
```
Comment (tsachiherman) [inline docker/build/Dockerfile-deploy:9]: maybe I'm missing something, but is that really related to network partitioning testing ?

File: test/testdata/deployednettemplates/recipes/network-partition/node.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+    "APIToken": "{{APIToken}}",
+    "EnableBlockStats": false,
+    "EnableTelemetry": false,
+    "TelemetryURI": "{{TelemetryURI}}",
+    "EnableMetrics": false,
+    "MetricsURI": "{{MetricsURI}}",
+    "ConfigJSONOverride": "{  \"TxPoolExponentialIncreaseFactor\": 1, \"DNSBootstrapID\": \"<network>.algodev.network\", \"DeadlockDetection\": -1, \"PeerPingPeriodSeconds\": 30, \"BaseLoggerDebugLevel\": 4, \"EnableProfiler\": true }",
+   "AltConfigs": [
```
Comment (btoll) [inline test/testdata/deployednettemplates/recipes/network-partition/node.json:9]: Nit: spacing

File: docker/build/Dockerfile-deploy (hunk 4 above)
Comment (egieseke) [reply docker/build/Dockerfile-deploy:9]: This fix is required to allow the `create_and_deploy_recipe.sh` script to function properly.

Comment (tsachiherman) [reply docker/build/Dockerfile-deploy:9]: was the `create_and_deploy_recipe.sh` broken ?
I'm pretty confident that your changes here doesn't affect the  `create_and_deploy_recipe.sh` directly, so I'm concerned about making changes that aren't needed.

If "direct" is broken, then we really need to figure why.

File: test/testdata/deployednettemplates/recipes/network-partition/gen_topology.py (hunk 3 above)
Comment (egieseke) [reply test/testdata/deployednettemplates/recipes/network-partition/gen_topology.py]: Removed.

File: docker/build/Dockerfile-deploy (hunk 4 above)
Comment (egieseke) [reply docker/build/Dockerfile-deploy:9]: Multipath GOProxy is not supported with go version 1.12.

File: netdeploy/remote/deployedNetwork.go (hunk 2 above)
Comment (egieseke) [reply netdeploy/remote/deployedNetwork.go]: Leave default as an empty string.

File: test/testdata/deployednettemplates/recipes/network-partition/node.json (hunk 5 above)
Comment (egieseke) [reply test/testdata/deployednettemplates/recipes/network-partition/node.json:9]: Removed extra white space.

File: docker/build/Dockerfile-deploy (hunk 4 above)
Comment (onetechnical) [reply docker/build/Dockerfile-deploy:9]: As we are going to shift to go 1.14 shortly, we may want to consider reverting this.

Comment (egieseke) [reply docker/build/Dockerfile-deploy]: Reverted back to multipath goproxy.

**Review by onetechnical — APPROVED** [review]
